### PR TITLE
feat(api+core+extension): telemetry enhancement

### PR DESCRIPTION
**Pull request by jayzhang**
1. add projectId in ProjectSettings
2. add correlationId in Inputs
3. core will generate a new projectId if it is not available
4. adjust UT related to ProjectSettings

### CODE REVIEW

**Review by hund030 — APPROVED** [review]

File: packages/cli/src/cmds/new.ts [hunk 1]
```diff
@@ -84,7 +84,9 @@ export default class New extends YargsCommand {
     const core = result.value;
 
     {
-      const result = await core.createProject(getSystemInputs());
+      const inputs = getSystemInputs();
+      inputs.projectId = uuid.v4();
```
Comment (chagong) [inline packages/cli/src/cmds/new.ts:88]: I thought it should core to generate the projectId. Should not be generated by client. And no need to put "projectId" property in Inputs.

File: packages/sdk/src/packageMetadata.ts [hunk 2]
```diff
@@ -2,4 +2,4 @@
 // Licensed under the MIT license.
 
 export const name = "@microsoft/teamsfx";
-export const version = "0.3.0-alpha.5";
+export const version = "0.3.0-alpha.6";
```
Comment (blackchoey) [inline packages/sdk/src/packageMetadata.ts:5]: Not blocking this PR.
@tecton seems the auto version bump does not commit changes made to packageMetadata.ts?

File: packages/fx-core/src/core/middleware/contextLoader.ts [hunk 3]
```diff
@@ -51,6 +52,8 @@ export async function loadSolutionContext(tools: Tools, inputs: Inputs):Promise<
     const projectSettings: ProjectSettings = await fs.readJson(settingsFile);
     if(!projectSettings.currentEnv)
       projectSettings.currentEnv = "default";
+    if(!projectSettings.projectId)
+      projectSettings.projectId = uuid.v4();
```
Comment (chagong) [inline packages/fx-core/src/core/middleware/contextLoader.ts:56]: when will it be written to env json file?

File: packages/cli/src/cmds/new.ts (hunk 1 above)
Comment (jayzhang) [reply packages/cli/src/cmds/new.ts:88]: ok

File: packages/fx-core/src/core/middleware/contextLoader.ts (hunk 3 above)
Comment (jayzhang) [reply packages/fx-core/src/core/middleware/contextLoader.ts:56]: in ConfigWriterMW

**Review by 1yefuwang1 — APPROVED** [review]

File: packages/vscode-extension/src/handlers.ts [hunk 4]
```diff
@@ -160,7 +161,8 @@ export function getSystemInputs():Inputs{
     projectPath: getWorkspacePath(),
     platform: Platform.VSCode,
     vscodeEnv: detectVsCodeEnv(),
-    "function-dotnet-checker-enabled": vscodeAdapter.dotnetCheckerEnabled()
+    "function-dotnet-checker-enabled": vscodeAdapter.dotnetCheckerEnabled(),
+    correlationId: uuid.v4()
```
Comment (1openwindow) [inline packages/vscode-extension/src/handlers.ts:165]: 1. how does the correlation Id pass to core?
2. echo user operation has different correlation id, how to handle it?

Comment (jayzhang) [reply packages/vscode-extension/src/handlers.ts:165]: 1. pass as property in Inputs object
2. Each operation has unique correlation id

File: packages/fx-core/src/common/telemetry.ts [hunk 5]
```diff
@@ -59,6 +61,11 @@ export function sendTelemetryEvent(
 			properties[TelemetryProperty.Component] = Component.cli;
 		}
 	}
+  
+  const correlationId = inputs.correlationId;
+  if(correlationId)
```
Comment (chagong) [inline packages/fx-core/src/common/telemetry.ts:66]: Make correlationId a mandatory property, If correlationId is empty, just put empty value for the property. Same for below.

**Review by IvanJobs — APPROVED** [review]

File: packages/api/src/types.ts [hunk 6]
```diff
@@ -162,6 +163,8 @@ export interface Inputs extends Json{
   ignoreLock?:boolean;
   ignoreTypeCheck?:boolean;
   ignoreConfigPersist?:boolean;
+  projectId?:string;
```
Comment (chagong) [inline packages/api/src/types.ts:166]: No need  to have it in Inputs.

**Review by chagong — APPROVED** [review]

**Review by 1openwindow — APPROVED** [review]

**Pull request merged**